src/policy/fees.cpp

```diff
@@ -133,8 +133,88 @@ class TxConfirmStats
      * variables with this state.
      */
     void Read(CAutoFile& filein, int nFileVersion, size_t numBuckets);
+
+    /**
+     * Helper function to check that TxConfirmStats behavior is consistent
+     */
+    int CheckConsistent(size_t expected_bucket_count) const;
+    int CheckConsistent(const std::vector<double>& exp_buckets, const std::map<double, unsigned int>& exp_bucket_map) const;
+
+    /**
+     * Copy state, adjusting stats for offset buckets
+     */
+    void ExtendStatsBelow(size_t added_buckets);
 };
 
+bool CBlockPolicyEstimator::IsConsistent() const
+{
+    LOCK(m_cs_fee_estimator);
+    return 0 == CheckConsistent();
+}
+
+int CBlockPolicyEstimator::CheckConsistent() const
+{
+    int res;
+    res = feeStats->CheckConsistent(buckets, bucketMap);
+    res += 32 * shortStats->CheckConsistent(buckets, bucketMap);
+    res += (32*32) * longStats->CheckConsistent(buckets, bucketMap);
+    return res;
+}
+
+int TxConfirmStats::CheckConsistent(size_t expected_bucket_count) const
+{
+    if (decay <= 0 || 1 <= decay) return 1;
```

Comment:
Might be helpful to wrap these return values in an `struct/class enum` with variants that describe the consistency conditions. You could then downcast back to an `int` when doing arithmetic like in `CBlockPlicyEstimator::CheckConsistent`.

Then, in the test cases, check the calls to `CheckConsistent` match the expected variant, instead of just being non-zero.

Reply:
They don't have any meaning, they're just distinct to allow you to refer back to the line that found that inconsistency.